src/main/scala/components/csr.scala

```diff
@@ -444,8 +444,15 @@ class CSRRegFile extends Module{
       reg_mstatus.sd := 0.U
     }
     
-    //MTVEC IS FIXED IN THIS IMPLEMENTATION
-
+    //MTVEC
+    /* write address to trap too
+     */
+    when (decoded_addr(MCSRs.mtvec)) {
+      val new_mtvec = wdata.asTypeOf(new MTVec())
```

Comment:
Can't you just say `reg_mtvec := wdata.asTypeOf(new MTVec())`?

Reply:
I suppose I could. I just wanted to be explicit in case of any weird optimization trickery. Should I change it?

Reply:
It's up to you.